dbus_dump_watcher.cpp

```diff
@@ -0,0 +1,141 @@
+#include "config.h"
+
+#include "dbus_dump_watcher.hpp"
+
+#include "dump_read_dbus_prop.hpp"
+#include "dump_send_pldm_cmd.hpp"
+
+#include <fmt/format.h>
+
+#include <phosphor-logging/log.hpp>
+
+namespace openpower::dump
+{
+using ::openpower::dump::utility::DBusInteracesMap;
+using ::openpower::dump::utility::DBusPropertiesMap;
+using ::openpower::dump::utility::ManagedObjectType;
+using ::phosphor::logging::level;
+using ::phosphor::logging::log;
+using ::sdbusplus::bus::match::rules::sender;
+
+DBusDumpWatcher::DBusDumpWatcher(sdbusplus::bus::bus& bus,
+                                 const std::string& entryIntf,
+                                 DumpType dumpType) :
+    _bus(bus),
+    _entryIntf(entryIntf), _dumpType(dumpType),
+    _intfAddWatch(std::make_unique<sdbusplus::bus::match_t>(
+        bus,
+        sdbusplus::bus::match::rules::interfacesAdded() + sender(dumpService),
+        [this](auto& msg) { this->interfaceAdded(msg); })),
+    _intfRemWatch(std::make_unique<sdbusplus::bus::match_t>(
+        bus,
+        sdbusplus::bus::match::rules::interfacesRemoved() + sender(dumpService),
+        [this](auto& msg) { this->interfaceRemoved(msg); }))
+{
+}
+
+void DBusDumpWatcher::interfaceAdded(sdbusplus::message::message& msg)
```

Comment:
we should not be handling every exception if so "std::string str" will also throw exception, we should handle if we have a recovery, or is expected like file open, now that I had done a catch in the main app it will go there and crash the application.

Reply:
I did not worry about general exceptions that's might occur very rarely.

Here,  we might get the exception from the sdbusplus libs read operation, I think that might throw an exception if there is an error code from the dbus broker but, Now you added the exception handling in the main and inside event loop so I believe the application will crash. 

Also considering the point, if one of the interfaces added signal failed to handle, we are crashing the app and we could not process other dump entry interface added signal but, I think that will be restored after this app reset.

Reply:
agree